### ISSUE TITLE
Settings are ignored using CLIEngine

### ISSUE THREAD

**Issue by thetalecrafter** [eslint/eslint#4402]
I'm having trouble using the CLIEngine to create a cli tool for validating internationalization (via format-message).

I was able to create a plugin that works great when using the actual eslint cli, but cannot seem to getting `settings` to get through to the rules using `CLIEngine`. I have tried putting `settings` on both the root options object, as well as inside a `baseConfig` object, and either way `context.settings` is always an empty object inside my rules. I have not tried using `configFile` (which likely works), because the settings are dynamic, and I'd rather avoid writing a temp file just to pass settings.

Is there a way to pass the desired `settings` for use in `cli.executeOnText`?

Here is roughly the code I am using:
```
var cli = new CLIEngine({
  plugins: [ 'format-message' ],
  rules: {
    'format-message/literal-locale': 1,
    'format-message/literal-pattern': 1,
    'format-message/no-identical-translation': 1,
    'format-message/no-invalid-pattern': 2,
    'format-message/no-invalid-translation': 2,
    'format-message/no-missing-params': [ 2, { 'allowNonLiteral': true } ],
    'format-message/no-missing-translation': 1,
    'format-message/translation-match-params': 2
  },
  settings: {
    'format-message': {
      translations: { en: { foo: 'bar' } }
    }
  }
})
cli.executeOnText(...)
...
```

**Comment by eslintbot** [comment]
Thanks for the issue! If you're reporting a bug, please be sure to include:

1. The version of ESLint you are using (run `eslint -v`)
2. What you did (the source code and ESLint configuration)
3. The actual ESLint output complete with numbers
4. What you expected to happen instead

Requesting a new rule? Please see [Proposing a New Rule](http://eslint.org/docs/developer-guide/contributing/new-rules) for instructions.

**Comment by thetalecrafter** [comment]
This is on eslint 1.9.0.

**Comment by gyandeeps** [comment]
Have you tried this:

```js
var cli = new CLIEngine({
    baseConfig: {
        plugins: [ 'format-message' ],
        rules: {
            'format-message/literal-locale': 1,
            'format-message/literal-pattern': 1,
            'format-message/no-identical-translation': 1,
            'format-message/no-invalid-pattern': 2,
            'format-message/no-invalid-translation': 2,
            'format-message/no-missing-params': [ 2, { 'allowNonLiteral': true } ],
            'format-message/no-missing-translation': 1,
            'format-message/translation-match-params': 2
        },
        settings: {
            'format-message': {
                translations: { en: { foo: 'bar' } }
            }
        }
    }
});
```

**Comment by thetalecrafter** [comment]
\o/ That gets settings through! I could have sworn I tried that, but perhaps I had a typo in there :confused:.

Thanks!

Now to figure out why I'm not seeing the lint errors I expect...

**Comment by arcanis** [comment]
Also got bitten by this issue - it's not really intuitive, maybe a `console.warn` would be helpful 😐
